AppSyncRealTimeClient/Support/TaskQueue.swift

```
@@ -0,0 +1,30 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#if swift(>=5.5.2)
+
+import Foundation
+
+@available(iOS 13.0, *)
+actor TaskQueue<Success> {
+    private var previousTask: Task<Success, Error>?
+
+    func sync(block: @Sendable @escaping () async throws -> Success) {
+        previousTask = Task { [previousTask] in
+            _ = await previousTask?.result
+            return try await block()
```

I _think_ this will swallow the thrown error. Could that be problematic here?